Duplicati/Library/Main/Database/LocalListDatabase.cs

```diff
@@ -367,13 +373,22 @@ public IEnumerable<IFileset> QuickSets
                                 @"SELECT DISTINCT ""ID"", ""IsFullBackup"" FROM ""Fileset"" ORDER BY ""Timestamp"" DESC ")
                         )
                         {
+                            // partial backup sets are only included until the first full backup is encountered
+                            var isFullBackupEncountered = false;
                             while (rd.Read())
                             {
                                 var id = rd.GetInt64(0);
-                                var isFullBackup = rd.GetInt32(1);
+                                var isFullBackup = rd.GetInt32(BackupType.FULL_BACKUP);

```

Comment:
Is the argument provided to `rd.GetInt32` supposed to be the ordinal of the column we are accessing?  Given the above query, I think this should be `rd.GetInt32(1)`, and the resulting value should be compared to `BackupType.FULL_BACKUP`.  Although this works since `BackupType.FULL_BACKUP = 1`, I don't think we should be providing `BackupType.FULL_BACKUP` as the argument to `rd.GetInt32`.

Together with the above comment, we would have something like
```
var isFullBackup = rd.GetInt32(1) == BackupType.FULL_BACKUP;
```